File: C++/3list/Square.cpp
/*
@author Stanisław Woźniak
@title: Square.cpp
*/
#include "Square.hpp"

Square::Square(double side){
  side1 = side;
  side2 = side;
  side3 = side;
  side4 = side;
  angle = 90;
}

double Square::field(){
  return side1*side1;
}

double Square::perimeter(){
  return 4*side1;
}

File: java/3list/src/com/stanislaw/quadrangles/Square.java
package com.stanislaw.quadrangles;
/*
@author: Stanisław Woźniak
@version: 1.0
*/
public class Square extends Quadrangle {

    public Square(double side1) {
        super(side1, side1, side1, side1, 90);
    }

    @Override
    public double field() {
        return this.side1*this.side1;
    }

    @Override
    public double perimeter() {
        return 4*this.side1;
    }
}
